### ISSUE TITLE
offline training 시 training 완료 시점 문의

### ISSUE THREAD

**Issue by blackaib** [Junyoungpark/Pytorch-AWAC#2]
일반적으로 PPO와 같은 강화학습 알고리즘의 loss 그래프를 supervised learning의 loss 그래프를 해석하듯이 보진 않는데,
offline learning에서는 학습이 잘되어가고 있다는 것을 loss의 지속적인 감소로 봐도 될까요?

AWAC exmample.ipynb에 작성된 offline learning 부분의 actor와 critic의 loss 그래프를 보면
마치 supervised learning의 학습 loss를 보듯이 해당 loss가 지속 낮아지는것을 확인 할 수 있었습니다.

오프라인 러닝에서는 신규 데이터 추가가 없고 정해진 데이터를 반복하여 학습하는 것이라
supervised 러닝에서처럼 loss 감소를 학습이 잘 되고 있는 것으로 해석해도 될 것 같아보이는데
AWAC exmample.ipynb의  actor와 critic의 loss 그래프를 통해 설명하고자 하시는 것이 그런 의미가 맞는지 문의드립니다.

그리고 코드에서는 배치사이즈와 업데이트수를 임의로 정하셨는데,
이런 수치를 어떤식으로 정하시는지도 궁금합니다.

**Comment by Junyoungpark** [comment]
- 적어도 논문에서 정한 두개의 loss metric (actor, critic) loss 가 잘 내려간다는것을 확인하려는 의도로 해당 그래프들을 배치하였습니다. 하지만 두개의 loss가 내려간다는게 꼭 궁극적인 RL의 control objective 를 개선시킨다는것은 조금 다른 이야기 일것같습니다.

- 말씀 주신 Hyperparameter 들외에 다른 하이퍼파라미터는 대부분은 `2^n` 의 가지게 설정하였거나, 몇몇 가지 실험을 해보고 결과가 잘 나오는 것으로 설정하였습니다.

---

For non-Korean speakers, the original issue said: (google translator-ed)

```In general, we do not look at the loss graph of reinforcement learning algorithms such as PPO as if we interpret the loss graph of supervised learning.
In offline learning, can it be seen that learning is going well as a continuous decrease in loss?

If you look at the loss graph of the actor and critic of the offline learning part written in AWAC exmample.ipynb
As if looking at the learning loss of supervised learning, we could confirm that the loss is continuously decreasing.

In offline learning, there is no addition of new data, and it is to learn by repeating set data.
As in supervised learning, it seems that the loss reduction can be interpreted as learning is going well.
I would like to ask if the meaning you are trying to explain through the loss graphs of actors and critics of AWAC exmample.ipynb is correct.

And in the code, the batch size and the number of updates were arbitrarily set.
I'm curious as to how you set these numbers.
```

- two loss curves (actor, critic) show the entire optimization process decreases the losses for training the actor and critic. However, the decrements of the losses don't mean the well-optimized actor improves the RL's control performance.

- All hyperparameters are set to be `2^n` whenever it is possible. Other hyperparameters given in the notebook were found after a few rounds of human hyperparameter tuning, i.e., trial-and-error.
